libs/mock-block-dock/src/datastore/use-mock-datastore.ts

```diff
@@ -567,89 +559,79 @@ export const useMockDatastore = (
             ],
           };
         }
-        // const { file, url, mediaType } = data;
-        // if (!file && !url?.trim()) {
-        //   throw new Error(
-        //     `Please enter a valid ${mediaType} URL or select a file below`,
-        //   );
-        // }
-        //
-        // if (url?.trim()) {
-        //   const resp = await createEntity({
-        //     data: {
-        //       entityTypeId: "file1",
-        //       properties: {
-        //         url,
-        //         mediaType,
-        //       },
-        //     },
-        //   });
-        //   if (resp.errors || !resp.data) {
-        //     return {
-        //       errors: resp.errors ?? [
-        //         {
-        //           code: "INVALID_INPUT",
-        //           message: "Could not create File entity ",
-        //         },
-        //       ],
-        //     };
-        //   }
-        //   const returnData: UploadFileReturn = {
-        //     entityId: resp.data.entityId,
-        //     mediaType,
-        //     url,
-        //   };
-        //   return Promise.resolve({ data: returnData });
-        // } else if (file) {
-        //   const result = await new Promise<FileReader["result"] | null>(
-        //     (resolve, reject) => {
-        //       const reader = new FileReader();
-        //
-        //       reader.onload = (event) => {
-        //         resolve(event.target?.result ?? null);
-        //       };
-        //
-        //       reader.onerror = (event) => {
-        //         reject(event);
-        //       };
-        //
-        //       reader.readAsDataURL(file);
-        //     },
-        //   );
-        //
-        //   if (result) {
-        //     const resp = await createEntity({
-        //       data: {
-        //         entityTypeId: "file1",
-        //         properties: {
-        //           url: result.toString(),
-        //           mediaType,
-        //         },
-        //       },
-        //     });
-        //     if (resp.errors || !resp.data) {
-        //       return {
-        //         errors: resp.errors ?? [
-        //           {
-        //             code: "INVALID_INPUT",
-        //             message: "Could not create File entity ",
-        //           },
-        //         ],
-        //       };
-        //     }
-        //     const returnData: UploadFileReturn = {
-        //       entityId: resp.data.entityId,
-        //       mediaType,
-        //       url: result.toString(),
-        //     };
-        //     return Promise.resolve({ data: returnData });
-        //   }
-        //
-        //   throw new Error("Couldn't read your file");
-        // }
-        // throw new Error("Unreachable.");
+        const { description } = data;
+        const file = "file" in data ? data.file : null;
```

Comment:
I do wonder if we should design the type to avoid us needing an `in` check, really don't like leaving the support of TS

Reply:
Added typeguards in https://github.com/blockprotocol/blockprotocol/pull/944/commits/8d53e9ca352a899dc85521206bebc44072dd72a1